doc/api/n-api.md

```diff
@@ -3022,6 +3023,102 @@ support it:
+on the native side, because the promise that has been derived from it can only
+be resolved or rejected by calling `napi_conclude_deferred()`.
+
+### napi_create_deferred
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status napi_create_promise(napi_env env,
+                                            napi_value* deferred,
+                                            napi_value* promise);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[out] deferred`: The deferred object.
+- `[out] promise`: The JavaScript promise associated with the deferred object.
+If `promise` is `NULL`, you can retrieve the associated JavaScript promise
+later using `napi_get_promise_from_deferred()`.
+
+Returns `napi_ok` if the API succeeded.
+
+This API creates a deferred object. Although this object is a JavaScript value
+it should not be passed into JavaScript because it lacks the functionality of
+a JavaScript promise. However, a JavaScript promise can be derived from it
+using `napi_get_promise_from_deferred()`. The derived promise can later be
+resolved or rejected by passing the deferred object to
+`napi_conclude_deferred()`.
+
+### napi_get_promise_from_deferred
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status napi_get_promise_from_deferred(napi_env env,
+                                                       napi_value deferred,
+                                                       napi_value* promise);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] deferred`: The deferred object from which to derive a promise.
+- `[out] promise`: The resulting promise.
+
+Returns `napi_ok` if the API succeeded.
+
+This API derives a JavaScript promise from a deferred object. Unlike the
+deferred object, the promise can be returned to JavaScript.
+
+### napi_conclude_deferred
+<!-- YAML
+added: REPLACEME
+-->
+```C
+NAPI_EXTERN napi_status napi_conclude_deferred(napi_env env,
+                                               napi_value deferred,
+                                               napi_value conclusion,
+                                               bool is_resolution);
+```
+
+- `[in] env`: The environment that the API is invoked under.
+- `[in] deferred`: The deferred object whose associated promise to conclude.
+- `[in] conclusion`: The value with which to conclude the promise.
+- `[in] is_rejection`: Flag indicating whether `conclusion` is a resolution
+(`true`) or a rejection (`false`).
+
+This API resolves or rejects a JavaScript promise by way of the deferred object
+from which it was derived. Thus, it can only be used to conclude JavaScript
+promises for which the corresponding deferred object is available. This
+effectively means that the promise must have been created using
+`napi_create_promise()` and the deferred object returned from that call must
+have been retained in order to be passed to this API.
+
+### napi_is_promise
```

Comment:
There's no corresponding `IsResolver()` in V8. I could mark it with some funky property, but that would be duck typing.